File: frontend/types/analysis.ts
export type Bias = "left" | "center" | "right";
export type Sentiment = "positive" | "neutral" | "negative";

export interface AnalysisResult {
  id: number;
  bias: Bias;
  bias_score: number;
  sentiment: Sentiment;
  sentiment_score: number;
  created_at: string;
}

export interface AnalyzeRequest {
  text: string;
}

File: backend/app/schemas/analysis.py
from datetime import datetime
from pydantic import BaseModel


class AnalyzeRequest(BaseModel):
    text: str


class AnalysisResult(BaseModel):
    id: int
    bias: str
    bias_score: float
    sentiment: str
    sentiment_score: float
    created_at: datetime

    model_config = {"from_attributes": True}
